### PR TITLE
properly recognizing existing required graph inputs

### PR DESCRIPTION
**Pull request by danbills**
does solve tests passing.

does not solve `getOrElse` part of ungrossify ticket

### CODE REVIEW

File: cwl/src/main/scala/wdl4s/cwl/WorkflowStep.scala [hunk 1]
```diff
@@ -103,51 +103,58 @@ case class WorkflowStep(
...
+                node <- set
+                outputPort <- node.outputPorts
+              } yield outputPort
 
-                val lookupSet: Set[OutputPort] =
-                  for {
-                    node <- set
-                    outputPort <- node.outputPorts
-                  } yield outputPort
+            lookupSet.find(_.name == inputSource)
```
Comment (mcovarr) [inline cwl/src/main/scala/wdl4s/cwl/WorkflowStep.scala:120]: mebbe put this as a filter in the forcomp?

File: cwl/src/main/scala/wdl4s/cwl/WorkflowStep.scala [hunk 2]
```diff
@@ -103,51 +103,58 @@ case class WorkflowStep(
...
-                    lookup == stepId
-                  }.get
+          FullyQualifiedName(inputSource) match {
+            case _: WorkflowInputId =>
+
+              val foundInSeenNodes = findThisInputInSet(knownNodes)
 
-                val upstreamNodesForFoundStep:Set[GraphNode] = step.callWithInputs(typeMap, workflow, knownNodes,workflowInputs)
+              val newNodesAndOutputPort: Option[OutputPort] =
+                foundInSeenNodes
```
Comment (mcovarr) [inline cwl/src/main/scala/wdl4s/cwl/WorkflowStep.scala:142]: inline

**Comment by cjllanwarne** [comment]
The `getOrElse` can-kicking notwithstanding... 👍 

[Approved with PullApprove](https://pullapprove.com/broadinstitute/wdl4s/pull-request/190/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/wdl4s)

**Comment by danbills** [comment]
The issues were unrelated so this is me learning my lesson and submitting separate PRs

**Review by mcovarr — APPROVED** [review]
Just some totally ignorable nitpicks, glad you got this working with a minimum of fuss!

**Comment by danbills** [comment]
@mcovarr whoops I merged before fixing stuff, I'll sneak your recommendations into the `getOrElse` fix PR

**Pull request merged**